File: app/src/main/java/jp/hazuki/yuzubrowser/action/item/startactivity/StartActivityPreferenceFragment.java
/*
 * Copyright (C) 2017 Hazuki
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package jp.hazuki.yuzubrowser.action.item.startactivity;

import android.content.Context;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.ListFragment;
import android.view.View;
import android.widget.ArrayAdapter;
import android.widget.ListAdapter;
import android.widget.ListView;

import jp.hazuki.yuzubrowser.R;

public class StartActivityPreferenceFragment extends ListFragment {

    private OnActionListener mListener;

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);

        ListAdapter adapter = new ArrayAdapter<>(getActivity(), android.R.layout.simple_list_item_1, getResources().getStringArray(R.array.action_start_activity_template));
        setListAdapter(adapter);
    }

    @Override
    public void onListItemClick(ListView l, View v, int position, long id) {
        switch (position) {
            case 0://Application
                if (mListener != null)
                    mListener.openApplicationList();
                break;
            case 1://Shortcut
                if (mListener != null)
                    mListener.openShortCutList();
                break;
            case 2://Share page
                if (mListener != null)
                    mListener.openSharePage();
                break;
            case 3://Open in other app
                if (mListener != null)
                    mListener.openOther();
                break;
        }
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        try {
            mListener = (OnActionListener) getActivity();
        } catch (ClassCastException e) {
            throw new IllegalStateException(e);
        }
    }

    @Override
    public void onDetach() {
        super.onDetach();
        mListener = null;
    }

    public interface OnActionListener {
        void openApplicationList();

        void openShortCutList();

        void openSharePage();

        void openOther();
    }
}

File: legacy/src/main/java/jp/hazuki/yuzubrowser/legacy/action/item/startactivity/StartActivityPreferenceFragment.kt
/*
 * Copyright (C) 2017-2019 Hazuki
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package jp.hazuki.yuzubrowser.legacy.action.item.startactivity

import android.content.Context
import android.os.Bundle
import android.view.View
import android.widget.ArrayAdapter
import android.widget.ListView
import jp.hazuki.yuzubrowser.legacy.R

class StartActivityPreferenceFragment : androidx.fragment.app.ListFragment() {

    private var mListener: OnActionListener? = null

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)

        listAdapter = ArrayAdapter(activity, android.R.layout.simple_list_item_1, resources.getStringArray(R.array.action_start_activity_template))
    }

    override fun onListItemClick(l: ListView, v: View, position: Int, id: Long) {
        when (position) {
            0 -> mListener?.openApplicationList() //Application
            1 -> mListener?.openShortCutList() //Shortcut
            2 -> mListener?.openSharePage() //Share page
            3 -> mListener?.openOther() //Open in other app
        }
    }

    override fun onAttach(context: Context) {
        super.onAttach(context)
        try {
            mListener = activity as OnActionListener
        } catch (e: ClassCastException) {
            throw IllegalStateException(e)
        }

    }

    override fun onDetach() {
        super.onDetach()
        mListener = null
    }

    interface OnActionListener {
        fun openApplicationList()

        fun openShortCutList()

        fun openSharePage()

        fun openOther()
    }
}
